extensions/VisualEditor/wikia/modules/ve/ui/dialogs/ve.ui.WikiaImageInsertDialog.js

```diff
@@ -0,0 +1,230 @@
+/*!
```

Comment:
My intention here was to do what you suggested: I use inheritClass to make subclasses of WikiaMediaInsertDialog. If you were suggesting some other type of subclassing, you'll have to clarify.

For example, WikiaImageInsertDialog inherits from the original media dialog:
`
OO.inheritClass( ve.ui.WikiaImageInsertDialog, ve.ui.WikiaMediaInsertDialog );
`

I override only these two functions:
ve.ui.WikiaImageInsertDialog.prototype.initialize
ve.ui.WikiaImageInsertDialog.prototype.insertPermanentMediaCallback

And these are the only ones I copy, the other functions are used directly from WikiaMediaInsertDialog and are not copied.

I do however copy all the .static. variables in the top part of the class: I checked oo.inheritClass and I see this at the end:

`	// Extend static properties - always initialize both sides
	oo.initClass( originFn );
	targetFn.static = Object.create( originFn.static );
`

so I think that means that the static properties are copied too (I had assumed only .prototype. was copied like what happens when you use JS new). I will try and remove the redundant .static. variables and see if it still works as intended.

(PS: I will address your other comments after the weekend)